arrivals/src/main/scala/com/pagerduty/arrivals/impl/filter/CombinableRequestFilter.scala

```diff
@@ -0,0 +1,21 @@
+package com.pagerduty.arrivals.impl.filter
+
+import akka.http.scaladsl.model.HttpRequest
+import com.pagerduty.arrivals.api
+import com.pagerduty.arrivals.api.filter.{RequestFilter, RequestFilterOutput}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait CombinableRequestFilter[-RequestData] extends api.filter.RequestFilter[RequestData] { base =>
+
+  def combine[U <: RequestData](filter: RequestFilter[U])(implicit ec: ExecutionContext): RequestFilter[U] = {
+    new RequestFilter[U] {
+      override def apply(request: HttpRequest, data: U): RequestFilterOutput = {
+        base.apply(request, data) flatMap {
```

Comment:
No-dot "infix" notation is generally discouraged for non-symbolic methods. See https://docs.scala-lang.org/style/method-invocation.html.
```suggestion
        base.apply(request, data).flatMap {
```

Reply:
Good to know, thanks!